Update to 0.30 changelog (#3907)
PennyLaneAI/pennylane: doc/releases/changelog-dev.md

<h3>New features since last release</h3>

<h4>Pulse programming</h4>

> * Added the needed functions and classes to simulate an ensemble of Rydberg atoms:

Reviewer: Need some code examples here for sure

Author: This is more placeholder for now, it needs to be replaced with a more catchy description of what's been done.

    the Hamiltonian of the interaction between a driving laser field and a group of atoms.
  [(#3749)](https://github.com/PennyLaneAI/pennylane/pull/3749)

> * Added `max_distance` keyword argument to `qml.pulse.rydberg_interaction` to allow removal of negligible contributions

Reviewer: Does this belong in the bulleted list above?

  from atoms beyond `max_distance`from each other.
  [(#3889)](https://github.com/PennyLaneAI/pennylane/pull/3889)

<h3>Improvements 🛠</h3>

<h4>Next generation device API</h4>

* The `sample_state` function is added to `devices/qubit` that returns a series of samples based on a given

>   state vector and a number of shots.

Reviewer: 🤔... How is this different from qml.sample?

Reviewer: Maybe a code example would help

  [(#3720)](https://github.com/PennyLaneAI/pennylane/pull/3720)

> * The `simulate` function added to `devices/qubit` now supports measuring expectation values of large observables such as

Reviewer: `devices/qubit` is not very clear. Does this mean every qubit device?

  `qml.Hamiltonian`, `qml.SparseHamiltonian`, `qml.Sum`.
  [(#3759)](https://github.com/PennyLaneAI/pennylane/pull/3759)

> * The `apply_operation` function added to `devices/qubit` now supports broadcasting.

Reviewer: see above

  [(#3852)](https://github.com/PennyLaneAI/pennylane/pull/3852)

> * `pennylane.devices.qubit.preprocess` now allows circuits with non-commuting observables.

Reviewer: 🤔

  [(#3930)](https://github.com/PennyLaneAI/pennylane/pull/3930)
  [(#3936)](https://github.com/PennyLaneAI/pennylane/pull/3936/)
  [(#3966)](https://github.com/PennyLaneAI/pennylane/pull/3966)

>   * Added `max_distance` keyword argument to `qml.pulse.rydberg_interaction` to allow removal of negligible contributions

Reviewer: ```suggestion
  * A new keyword argument called `max_distance` has been added to `qml.pulse.rydberg_interaction` to allow for the removal of negligible contributions
```

>     from atoms beyond `max_distance`from each other.

Reviewer: ```suggestion
    from atoms beyond `max_distance` from each other.
```

    [(#3889)](https://github.com/PennyLaneAI/pennylane/pull/3889)

<h4>Quantum singular value transform</h4>

<h4>Intuitive QNode returns</h4>

> * The new return system is activated and public facing. The qnode kwarg `mode` is replaced by the boolean 

Reviewer: ```suggestion
* The new return system is now activated and public facing. The QNode keyword argument `mode` is replaced by the boolean  
```

>   `grad_on_execution` .

Reviewer: ```suggestion
  `grad_on_execution`.
```

  [(#3957)](https://github.com/PennyLaneAI/pennylane/pull/3957)
  [(#3969)](https://github.com/PennyLaneAI/pennylane/pull/3969)

<h3>Improvements 🛠</h3>

<h4>Next generation device API</h4>

* The `sample_state` function is added to `devices/qubit` that returns a series of samples based on a given
  state vector and a number of shots.
  [(#3720)](https://github.com/PennyLaneAI/pennylane/pull/3720)

* The `simulate` function added to `devices/qubit` now supports measuring expectation values of large observables such as
  `qml.Hamiltonian`, `qml.SparseHamiltonian`, `qml.Sum`.
  [(#3759)](https://github.com/PennyLaneAI/pennylane/pull/3759)

* The `apply_operation` function added to `devices/qubit` now supports broadcasting.
  [(#3852)](https://github.com/PennyLaneAI/pennylane/pull/3852)

* `pennylane.devices.qubit.preprocess` now allows circuits with non-commuting observables.
  [(#3857)](https://github.com/PennyLaneAI/pennylane/pull/3857)

* Adjoint differentiation support for the new qubit state-vector device has been added via
  `adjoint_jacobian` in `devices/qubit`.
  [(#3790)](https://github.com/PennyLaneAI/pennylane/pull/3790)

> * Adds logic to `qml.devices.qubit.measure` to compute the expectation values of `Hamiltonian` and `Sum`
>   in a backpropagation compatible way.

Reviewer: ```suggestion
* qml.devices.qubit.measure` now computes the expectation values of `Hamiltonian` and `Sum`
  in a backpropagation-compatible way.
```

  [(#3939)](https://github.com/PennyLaneAI/pennylane/pull/3939)

* The adjoint differentiation method now supports more operations, and does no longer decompose
  some operations that may be differentiated directly. In addition, all new operations with a
  generator are now supported by the method.
  [(#3874)](https://github.com/PennyLaneAI/pennylane/pull/3874)

* When using Jax-jit with gradient transforms the trainable parameters are correctly set (instead of every parameter 
  to be set as trainable), and therefore the derivatives are computed more efficiently.
  [(#3697)](https://github.com/PennyLaneAI/pennylane/pull/3697)

* `CompositeOp` now overrides `Operator._check_batching`, providing a significant performance improvement.
  `Hamiltonian` also overrides this method and does nothing, because it does not support batching.
  [(#3915)](https://github.com/PennyLaneAI/pennylane/pull/3915)

* If a `Sum` operator has a pre-computed Pauli representation, `is_hermitian` now checks that all coefficients
  are real, providing a significant performance improvement.
  [(#3915)](https://github.com/PennyLaneAI/pennylane/pull/3915)

> * 3 new decomposition algorithms are added for n-controlled operations with single-qubit target,

Reviewer: ```suggestion
* Three new decomposition algorithms are added for n-controlled operations with a single-qubit target
```

  [(#3936)](https://github.com/PennyLaneAI/pennylane/pull/3936/)
  [(#3966)](https://github.com/PennyLaneAI/pennylane/pull/3966)
  [(#3987)](https://github.com/PennyLaneAI/pennylane/pull/3987)
  * A new keyword argument called `max_distance` has been added to `qml.pulse.rydberg_interaction` to allow for the removal of negligible contributions
    from atoms beyond `max_distance` from each other.
    [(#3889)](https://github.com/PennyLaneAI/pennylane/pull/3889)

> <h4>Quantum singular value transform</h4>

Reviewer 2: This is a placeholder?

<h4>Intuitive QNode returns</h4>

> * The new return system is now activated and public facing. The QNode keyword argument `mode` is replaced by the boolean  

Reviewer 2: ```suggestion
* The new return system is now activated and public-facing. The QNode keyword argument `mode` is replaced by the boolean  
```

  `grad_on_execution`.
  [(#3957)](https://github.com/PennyLaneAI/pennylane/pull/3957)
  [(#3969)](https://github.com/PennyLaneAI/pennylane/pull/3969)

<h3>Improvements 🛠</h3>

> <h4>Next generation device API</h4>

Reviewer 2: ```suggestion
<h4>Next-generation device API</h4>
```

* The `sample_state` function is added to `devices/qubit` that returns a series of samples based on a given
  state vector and a number of shots.
  [(#3720)](https://github.com/PennyLaneAI/pennylane/pull/3720)

* The `simulate` function added to `devices/qubit` now supports measuring expectation values of large observables such as
  `qml.Hamiltonian`, `qml.SparseHamiltonian`, `qml.Sum`.
  [(#3759)](https://github.com/PennyLaneAI/pennylane/pull/3759)

* The `apply_operation` function added to `devices/qubit` now supports broadcasting.
  [(#3852)](https://github.com/PennyLaneAI/pennylane/pull/3852)

* `pennylane.devices.qubit.preprocess` now allows circuits with non-commuting observables.
  [(#3857)](https://github.com/PennyLaneAI/pennylane/pull/3857)

* Adjoint differentiation support for the new qubit state-vector device has been added via
  `adjoint_jacobian` in `devices/qubit`.
  [(#3790)](https://github.com/PennyLaneAI/pennylane/pull/3790)

> * qml.devices.qubit.measure` now computes the expectation values of `Hamiltonian` and `Sum`

Reviewer 2: ```suggestion
* `qml.devices.qubit.measure` now computes the expectation values of `Hamiltonian` and `Sum`
```

  in a backpropagation-compatible way.
  [(#3862)](https://github.com/PennyLaneAI/pennylane/pull/3862/)

<h4>Performance improvements</h4>

* Added a new decomposition to `qml.SingleExcitation` that halves the number of
  CNOTs required.
  [(3976)](https://github.com/PennyLaneAI/pennylane/pull/3976)

> * Improve the efficiency of `tapering()`, `tapering_hf()` and `clifford()`.

Reviewer 2: ```suggestion
* Improved efficiency of `tapering()`, `tapering_hf()` and `clifford()`.
```

  [(3942)](https://github.com/PennyLaneAI/pennylane/pull/3942)

> * Update Pauli arithmetic to more efficiently convert to a Hamiltonian.

Reviewer 2: ```suggestion
* Updated Pauli arithmetic to more efficiently convert to a Hamiltonian.
```

  [(#3939)](https://github.com/PennyLaneAI/pennylane/pull/3939)

* The adjoint differentiation method now supports more operations, and does no longer decompose
  some operations that may be differentiated directly. In addition, all new operations with a
  generator are now supported by the method.
  [(#3874)](https://github.com/PennyLaneAI/pennylane/pull/3874)

> * When using Jax-jit with gradient transforms the trainable parameters are correctly set (instead of every parameter 

Reviewer 2: ```suggestion
* When using `jax.jit` with gradient transforms, the trainable parameters are set correctly (instead of every parameter having
```
→ Alternatively, you could use "JAX-JIT"
→ I hope I understood right what you were trying to say in the middle of the sentence, I tried rephrasing it

  to be set as trainable), and therefore the derivatives are computed more efficiently.
  [(#3697)](https://github.com/PennyLaneAI/pennylane/pull/3697)

* `CompositeOp` now overrides `Operator._check_batching`, providing a significant performance improvement.
  `Hamiltonian` also overrides this method and does nothing, because it does not support batching.
  [(#3915)](https://github.com/PennyLaneAI/pennylane/pull/3915)

* If a `Sum` operator has a pre-computed Pauli representation, `is_hermitian` now checks that all coefficients
  are real, providing a significant performance improvement.
  [(#3915)](https://github.com/PennyLaneAI/pennylane/pull/3915)

> * Three new decomposition algorithms are added for n-controlled operations with a single-qubit target

Reviewer 2: ```suggestion
* Three new decomposition algorithms have been added for n-controlled operations with a single-qubit target
```

  and are selected automatically when they produce a better result, i.e., fewer CNOT gates.
  They can be accessed via `ops.op_math.ctrl_decomp_bisect`.
  [(#3851)](https://github.com/PennyLaneAI/pennylane/pull/3851)

<h4>Intuitive QNode returns</h4>

> * The default gaussian device and parameter shift cv support the new return system but only for single measurement.

Reviewer 2: ```suggestion
* The default Gaussian device and parameter shift CV support the new return system, but only for single measurements.
```
Generally preferred to capitalize 'Gaussian' because it's a proper name, but NBD because not everyone follows that rule.

  [(3946)](https://github.com/PennyLaneAI/pennylane/pull/3946)

* Keras and Torch NN modules are now compatible with the new return type system.
  [(#3913)](https://github.com/PennyLaneAI/pennylane/pull/3913)
  [(#3914)](https://github.com/PennyLaneAI/pennylane/pull/3914)

* `DefaultQutrit` supports the new return system.
  [(#3934)](https://github.com/PennyLaneAI/pennylane/pull/3934)

> * The default gaussian device and parameter shift cv support the new return system but only for single measurement.

Reviewer 2: ```suggestion
```
Repeated from a few lines above.

  in the circuit.
  [(#3888)](https://github.com/PennyLaneAI/pennylane/pull/3888)

* 3 new decomposition algorithms are added for n-controlled operations with single-qubit target,

>   and are selected automatically when they produce a better result. They can be accessed via

Reviewer: What does "better result" mean?

  `ops.op_math.ctrl_decomp_bisect`.
  [(#3851)](https://github.com/PennyLaneAI/pennylane/pull/3851)

> * Added `Operation.__truediv__` dunder method to be able to divide operators.

Reviewer: ➗

  [(#3749)](https://github.com/PennyLaneAI/pennylane/pull/3749)

<h3>Breaking changes 💔</h3>

> * Both JIT interfaces are not compatible with Jax `>0.4.3`, we raise an error for those versions.

Reviewer: Just making sure, but we're saying to users that they need to use `<= 0.4.3 ` ?

* `AdaptiveOptimizer` is updated to use non-default user-defined qnode arguments.
  [(#3765)](https://github.com/PennyLaneAI/pennylane/pull/3765)

* Use `TensorLike` type in `Operator` dunder methods.
  [(#3749)](https://github.com/PennyLaneAI/pennylane/pull/3749)

> * `qml.QubitStateVector.state_vector` now supports broadcasting.

Reviewer: Points akin to this one (i.e., X now supports broadcasting, or X supports the new return type system) can probably be put into one single point that lists off all of the X's

> * Do not run `qml.transforms.sum_expand` in `Device.batch_transform` if the device supports Sum observables.

Reviewer 2: ```suggestion
* `qml.transforms.sum_expand` is not run in `Device.batch_transform` if the device supports Sum observables.
```

  [(#3915)](https://github.com/PennyLaneAI/pennylane/pull/3915)

> * The type of `n_electrons` in `qml.qchem.Molecule` is set to `int`.

Reviewer 2: ```suggestion
* The type of `n_electrons` in `qml.qchem.Molecule` has been set to `int`.
```

  [(#3885)](https://github.com/PennyLaneAI/pennylane/pull/3885)

* Added explicit errors to `QutritDevice` if `classical_shadow` or `shadow_expval` are measured.
  [(#3934)](https://github.com/PennyLaneAI/pennylane/pull/3934)

* `QubitDevice` now defines the private `_get_diagonalizing_gates(circuit)` method and uses it when executing circuits.
  This allows devices that inherit from `QubitDevice` to override and customize their definition of diagonalizing gates.
  [(#3938)](https://github.com/PennyLaneAI/pennylane/pull/3938)

> * `retworkx` has been renamed to `rustworkx` to accommodate the change in name for the package.

Reviewer 2: ```suggestion
* `retworkx` has been renamed to `rustworkx` to accommodate the change in the package name.
```

  ambiguity between `op.num_wires = 0` and `op.num_wires = AllWires`.
  [(#3978)](https://github.com/PennyLaneAI/pennylane/pull/3978)

> <h3>Breaking changes 💔</h3>

Reviewer 2: Lovely change. :D

> * Both JIT interfaces are not compatible with Jax `>0.4.3`, we raise an error for those versions.

Reviewer 2: ```suggestion
* Both JIT interfaces are not compatible with JAX `>0.4.3`, we raise an error for those versions.
```

  [(#3958)](https://github.com/PennyLaneAI/pennylane/pull/3958)
  [(#3983)](https://github.com/PennyLaneAI/pennylane/pull/3983)

> <h3>Deprecations 👋</h3>

Reviewer 2: If this stays empty, ideally add a simple sentence saying there's none. Or "(None.)"
